eng/ci/public-build.yml

```diff
@@ -34,7 +35,9 @@ resources:
     ref: refs/tags/release
 
 variables:
-  - template: /eng/ci/templates/variables/build.yml@self
+- template: /eng/ci/templates/variables/build.yml@self
+- name: buildNumber
+  value: $[ counter('build', 0) ]
```

Comment:
nit: do we have to change the indentation for all the variables? Everything else has an indentation so it's weird that this wouldn't

Reply:
this is making indentation consistent: when there is a `-`, we don't indent (because then the 'body' of the line is at the indentation we want). When there is no `-` we indent.

Stated differently: an indent is either `- ` (- plus a space), OR `  ` (two spaces)